foursight_core/react/api/react_api.py

```diff
@@ -0,0 +1,447 @@
+        response.headers = {"Content-Type": content_type}
+        response.status_code = 200
+        return response
+
+    def create_redirect_response(self, location: str, headers: dict):
+        if not headers:
+            headers = {}
+        headers["location"] = location
+        return Response(status_code=302, body=json.dumps(headers), headers=headers)
+
+    def is_react_authentication(self, auth0_response: dict) -> bool:
+        """
+        Returns True iff the given Auth0 authentication response (specifically, from the POST
+        to https://hms-dbmi.auth0.com/oauth/token which happens in app_utils.auth0_callback)
+        is for authentication from the React UI. This is communicated via an Auth0 "scope"
+        which is setup on the React UI side to contain a "react" string.
+        See: react/src/pages/LoginPage/createAuth0Lock.
+        """
+        return "react" in auth0_response.get("scope", "") if auth0_response else False
+
+    def react_authentication_callback(self, request: dict, env: str, jwt: str, jwt_expires_in: int, domain: str, context: str):
+        """
+        Called from the main Auth0 callback, in app_utils/auth0_callback, AFTER the Auth0 HTTP POST
+        which does the actual authentication; that POST returns the JWT which is received by this
+        function. So at this point, the user HAS been SUCCESSFULLY authenticated and we have a
+        VALID/authenticated JWT; if this were not so we would have returned before this call,
+        in app_utils/auth_callback. We create a new JWT from the given JWT, which we call authtoken,
+        and set this as a cookie on the redirect (HTTP 302) response which we return.
+        The jwt_expires_in is the number of seconds from now when the given JWT will expire;
+        this is directly from the (above mentioned) Auth0 POST; this is NOT the exp from
+        the JWT which seems to be different from (like 14 hours less than) jwt_expires_in.
+        """
+        jwt_expires_at = convert_datetime_to_time_t(datetime.datetime.utcnow() + datetime.timedelta(seconds=jwt_expires_in))
+        authtoken = self.auth.create_authtoken(jwt, jwt_expires_at, env, domain)
+        authtoken_cookie = create_set_cookie_string(request, name="authtoken",
+                                                    value=authtoken,
+                                                    domain=domain,
+                                                    expires=jwt_expires_at, http_only=False)
+        redirect_url = self._get_redirect_url(request, env, domain, context)
+        return self.create_redirect_response(redirect_url, {"set-cookie": authtoken_cookie})
+
+    def _get_redirect_url(self, request: dict, env: str, domain: str, context: str):
+        redirect_url = read_cookie(request, "reactredir")
+        if not redirect_url:
+            is_running_locally = app.core.is_running_locally(request)
+            if is_running_locally:
+                http = "http"
```

Comment:
Maybe `scheme` instead of `http` for variable name?

Reply:
Sounds good.